### ISSUE TITLE
`@stylistic/unit-case` reads a hash that opens no interpolation as part of the unit

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#426]
Found while writing the branch that closes [#298](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/298). Reproduces on `main` before that branch as it does after it, character for character. Plain CSS, no preprocessor and no comment.

### What minimal example or steps are needed to reproduce the bug?

A number, a unit, and a hash standing behind them in one word:

```css
a { b: 10px#fff; }
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "rules": {
    "@stylistic/unit-case": "upper"
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.css" --config myconfig.json --fix
```

### What did you expect to happen?

The unit reported as `px`, or the word passed over. `pxfff` is no unit, and the hash behind the dimension belongs to `color-hex-case` rather than to this rule.

### What actually happened?

```
1:10-1:16  Expected "pxfff" to be "PXFFF"
```

and `--fix` leaves `a { b: 10PX#FFF; }`, with the hash recased along with the unit. Under `lower` the same input comes back `a { b: 10px#fff; }` from `10PX#FFF`, so the two options move the hash in opposite directions.

### What is the cause?

`getDimension` cuts every run of `[#@{}]` out of the copy it reads, under the name `EVERY_INTERPOLATION_CHARACTER` and with the comment «remove non standard stuff». Those characters spell an interpolation only where an interpolation is written: here the hash opens nothing, and cutting it glues `10px` to `fff`, so `valueParser.unit` reads `pxfff` as the unit of `10`. The warning names that run, and the fix recases the whole word.

The `startsWith("#")` guard just above catches a word that *opens* with a hash and nothing else.

### Do you have a proposal to fix the bug?

Cut the characters of an interpolation where an interpolation stands, and nowhere else. `findInterpolationSpans`, which [#298](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/298) added, places them in the value, and a run of `[#@{}]` standing outside every one of those spans is a character of the word rather than of an interpolation — a word carrying one is either a dimension whose unit ends where that character begins, or no dimension at all.
